drivers/flash/nrf_qspi_nor.c

```diff
@@ -410,6 +428,69 @@ static int qspi_wait_while_writing(const struct device *dev)
 	return (ret < 0) ? ret : 0;
 }
 
+static int qspi_wrsrX(const struct device *dev, uint8_t sr_val, uint8_t sr_num)
+{
+	int ret = 0;
+	uint8_t opcode = SPI_NOR_CMD_WRSR;
+	uint8_t length = 1;
+	uint8_t sr_array[2] = {0};
+
+	if (sr_num == 1) {
+		// Writing sr1 clears sr2. Need to read/modify/write both.
+		sr_array[0] = sr_val;
+		if (INST_0_QER == JESD216_DW15_QER_S2B1v1) {
+			ret = qspi_rdsr2(dev);
```

Comment:
This is a bug, there is no `qspi_rdsr2` function declared. I think you mean `qspi_rdsr(dev, 2)` ?

Reply:
Yep. Meant to include a qspi_rdsr2 wrapper inline (like rdsr) and it slipped out of the commit. Fixed now